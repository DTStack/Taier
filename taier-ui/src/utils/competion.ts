/* eslint-disable no-template-curly-in-string */
// competion for monaco-editor
import { languages } from '@dtinsight/molecule/esm/monaco';

export const Keywords = (range: languages.CompletionItem['range']): languages.CompletionItem[] => [
	{
		label: 'SELECT',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The SELECT statement is used to select data from a database.',
		insertText: 'SELECT',
		range,
	},
	{
		label: 'CREATE TABLE',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The CREATE TABLE statement is used to create a new table in a database.',
		insertText: 'CREATE TABLE',
		range,
	},
	{
		label: 'WHERE',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The WHERE clause is used to filter records.',
		insertText: 'WHERE',
		range,
	},
	{
		label: 'AND',
		kind: languages.CompletionItemKind.Operator,
		documentation:
			'The AND operator is used to filter records based on more than one condition, which displays a record if all the conditions separated by AND are TRUE.',
		insertText: 'AND',
		range,
	},
	{
		label: 'OR',
		kind: languages.CompletionItemKind.Operator,
		documentation:
			'The OR operator is used to filter records based on more than one condition, which displays a record if any of the conditions separated by OR is TRUE.',
		insertText: 'OR',
		range,
	},
	{
		label: 'NOT',
		kind: languages.CompletionItemKind.Operator,
		documentation: 'The NOT operator displays a record if the condition(s) is NOT TRUE.',
		insertText: 'NOT',
		range,
	},
	{
		label: 'ORDER BY',
		kind: languages.CompletionItemKind.Keyword,
		documentation:
			'The ORDER BY keyword is used to sort the result-set in ascending or descending order. The ORDER BY keyword sorts the records in ascending order by default. To sort the records in descending order, use the DESC keyword.',
		insertText: 'ORDER BY',
		range,
	},
	{
		label: 'INSERT INTO',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The INSERT INTO statement is used to insert new records in a table.',
		insertText: 'INSERT INTO',
		range,
	},
	{
		label: 'NULL',
		kind: languages.CompletionItemKind.Value,
		documentation: 'A field with a NULL value is a field with no value.',
		insertText: 'NULL',
		range,
	},
	{
		label: 'UPDATE',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The UPDATE statement is used to modify the existing records in a table.',
		insertText: 'UPDATE',
		range,
	},
	{
		label: 'DELETE',
		kind: languages.CompletionItemKind.Keyword,
		documentation: 'The DELETE statement is used to modify the existing records in a table.',
		insertText: 'DELETE',
		range,
	},
];

export const Snippets = (range: languages.CompletionItem['range']): languages.CompletionItem[] => [
	{
		label: 'INSERT:snippet',
		kind: languages.CompletionItemKind.Snippet,
		documentation: 'The INSERT INTO statement is used to insert new records in a table.',
		insertText: 'INSERT INTO ${1:tableName} VALUES(${2:value});',
		insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
		range,
	},
	{
		label: 'CREATE:snippet',
		kind: languages.CompletionItemKind.Snippet,
		documentation: 'The CREATE TABLE statement is used to create a new table in a database.',
		insertText: 'CREATE TABLE ${1:table_name} (\n\t${2:column} ${3:datatype}\n);',
		insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
		range,
	},
	{
		label: 'SELECT:snippet',
		kind: languages.CompletionItemKind.Snippet,
		documentation: 'The SELECT statement is used to select data from a database.',
		insertText: 'SELECT ${1:column} FROM ${2:table_name};',
		insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
		range,
	},
	{
		label: 'SHOW:snippet',
		kind: languages.CompletionItemKind.Snippet,
		documentation: 'show tables;',
		insertText: 'show tables;',
		insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
		range,
	},
];
