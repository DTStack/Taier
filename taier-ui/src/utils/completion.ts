import { languages } from '@dtinsight/molecule/esm/monaco';

interface ICompletionProps {
    keyword: string;
    description: string;
}

let keywords: ICompletionProps[] = [];
let functions: ICompletionProps[] = [];
export const Keywords = async (range: languages.CompletionItem['range']): Promise<languages.CompletionItem[]> => {
    if (!keywords.length || !functions.length) {
        return Promise.all([
            fetch('./assets/keywords.json', { method: 'get' }).then<ICompletionProps[]>((res) => res.json()),
            fetch('./assets/functions.json', { method: 'get' }).then<ICompletionProps[]>((res) => res.json()),
        ])
            .then(([keywordList, functionList]) => {
                keywords = keywordList;
                functions = functionList;
            })
            .then(() => {
                return [
                    ...keywords.map((keyword) => ({
                        label: keyword.keyword,
                        kind: languages.CompletionItemKind.Keyword,
                        documentation: keyword.description,
                        insertText: keyword.keyword,
                        range,
                    })),
                    ...functions.map((fn) => ({
                        label: fn.keyword,
                        kind: languages.CompletionItemKind.Function,
                        documentation: fn.description,
                        insertText: fn.keyword,
                        range,
                    })),
                    {
                        label: 'NULL',
                        kind: languages.CompletionItemKind.Value,
                        documentation: 'A field with a NULL value is a field with no value.',
                        insertText: 'NULL',
                        range,
                    },
                ];
            });
    }

    return [
        ...keywords.map((keyword) => ({
            label: keyword.keyword,
            kind: languages.CompletionItemKind.Keyword,
            documentation: keyword.description,
            insertText: keyword.keyword,
            range,
        })),
        ...functions.map((fn) => ({
            label: fn.keyword,
            kind: languages.CompletionItemKind.Function,
            documentation: fn.description,
            insertText: fn.keyword,
            range,
        })),
        {
            label: 'NULL',
            kind: languages.CompletionItemKind.Value,
            documentation: 'A field with a NULL value is a field with no value.',
            insertText: 'NULL',
            range,
        },
    ];
};

export const Snippets = (range: languages.CompletionItem['range']): languages.CompletionItem[] => [
    {
        label: 'INSERT:snippet',
        kind: languages.CompletionItemKind.Snippet,
        documentation: 'The INSERT INTO statement is used to insert new records in a table.',
        insertText: 'INSERT INTO ${1:tableName} VALUES(${2:value});',
        insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
        range,
    },
    {
        label: 'CREATE:snippet',
        kind: languages.CompletionItemKind.Snippet,
        documentation: 'The CREATE TABLE statement is used to create a new table in a database.',
        insertText: 'CREATE TABLE ${1:table_name} (\n\t${2:column} ${3:datatype}\n);',
        insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
        range,
    },
    {
        label: 'SELECT:snippet',
        kind: languages.CompletionItemKind.Snippet,
        documentation: 'The SELECT statement is used to select data from a database.',
        insertText: 'SELECT ${1:column} FROM ${2:table_name};',
        insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
        range,
    },
    {
        label: 'SHOW:snippet',
        kind: languages.CompletionItemKind.Snippet,
        documentation: 'show tables;',
        insertText: 'show tables;',
        insertTextRules: languages.CompletionItemInsertTextRule.InsertAsSnippet,
        range,
    },
];
