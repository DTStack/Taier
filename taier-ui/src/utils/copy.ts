/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Copy工具类
 *
 * 使用方法：
 *  import CopyUtils from 'copy.js';
 *  const instance = new CopyUtils();
 *  instance.copy(value, callback);
 */
export default class CopyUtils {
  fakeHandlerCallback: any;
  fakeHandler: any = null;
  fakeElem: any;

  copy(value: any, callback?: Function) {
    this.removeFake();

    this.fakeHandlerCallback = () => this.removeFake();
    this.fakeHandler = document.body.addEventListener(
      'click',
      this.fakeHandlerCallback,
    );

    this.fakeElem = document.createElement('textarea');
    // Prevent zooming on iOS
    this.fakeElem.style.fontSize = '12pt';

    // Reset box model
    this.fakeElem.style.border = '0';
    this.fakeElem.style.padding = '0';
    this.fakeElem.style.margin = '0';

    // Move element out of screen horizontally
    this.fakeElem.style.position = 'absolute';
    this.fakeElem.style.left = '-9999px';

    // Move element to the same position vertically
    let yPosition = window.pageYOffset || document.documentElement.scrollTop;
    this.fakeElem.style.top = `${yPosition}px`;

    this.fakeElem.setAttribute('readonly', '');
    this.fakeElem.value = value;

    document.body.appendChild(this.fakeElem);
    this.fakeElem.select();

    callback && this.copyText(callback);
  }

  removeFake() {
    if (this.fakeHandler) {
      document.body.removeEventListener('click', this.fakeHandlerCallback);
      this.fakeHandler = null;
      this.fakeHandlerCallback = null;
    }

    if (this.fakeElem) {
      document.body.removeChild(this.fakeElem);
      this.fakeElem = null;
    }
  }

  copyText(callback: Function) {
    let succeeded;

    try {
      succeeded = document.execCommand('copy');
    } catch (err) {
      succeeded = false;
    }
    if (callback) callback(succeeded);
  }
}
