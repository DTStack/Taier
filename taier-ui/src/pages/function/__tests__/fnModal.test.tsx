import { fireConfirmOnModal } from '@/tests/utils';
import { cleanup, render, waitFor } from '@testing-library/react';
import { Input } from 'antd';
import FnModal from '../fnModal';
import { fillFormContent } from './fixtures/utils';

jest.useFakeTimers();
jest.mock('@/services/resourceManagerService', () => ({
    checkNotDir: jest.fn().mockResolvedValue(true),
}));
jest.mock('../../../components/folderPicker', () => ({ id, value, onChange }: any) => (
    <Input id={id} value={value} onChange={onChange} data-testid="mockFolderPicker" />
));
jest.mock('@/context', () => {
    const react = jest.requireActual('react');
    return react.createContext({
        supportJobTypes: [{ key: 1, value: 'spark' }],
    });
});

describe('Test FnModal Component for Function', () => {
    beforeEach(() => {
        cleanup();
    });
    it('Should match snapshots', () => {
        expect(render(<FnModal visible />).asFragment()).toMatchSnapshot('create new function');
        expect(
            render(
                <FnModal
                    visible
                    data={{
                        taskType: 1,
                        name: 'test',
                        id: 1,
                    }}
                />
            ).asFragment()
        ).toMatchSnapshot('edit function');
    });

    it('Should call onAddFunction when create a new function', async () => {
        const fn = jest.fn().mockResolvedValue(true);
        const onCloseFn = jest.fn();
        const { container, getByTestId, getAllByTestId } = render(
            <FnModal visible onAddFunction={fn} onClose={onCloseFn} />
        );
        const dom = document.createElement('div');
        dom.id = 'molecule';
        container.appendChild(dom);

        fillFormContent(getAllByTestId);

        fireConfirmOnModal(getByTestId);

        await waitFor(() => {
            expect(fn).toBeCalledWith({
                className: 'com.dtstack',
                commandFormate: 'main',
                name: 'mock',
                nodePid: '1',
                paramDesc: 'test',
                purpose: 'forTest',
                resourceId: '1',
                taskType: 5,
                udfType: 0,
            });
        });

        expect(onCloseFn).toBeCalled();
    });

    it('Should call onEditFunction when edit a function', async () => {
        const fn = jest.fn().mockResolvedValue(true);
        const onCloseFn = jest.fn();
        const { getByTestId } = render(
            <FnModal
                visible
                onEditFunction={fn}
                onClose={onCloseFn}
                data={{
                    id: 1,
                    className: 'com.dtstack',
                    commandFormate: 'main',
                    name: 'mock',
                    nodePid: 1,
                    paramDesc: 'test',
                    purpose: 'forTest',
                    resources: 1,
                    taskType: 5,
                    udfType: 0,
                }}
            />
        );

        fireConfirmOnModal(getByTestId);

        await waitFor(() => {
            expect(fn).toBeCalledWith({
                className: 'com.dtstack',
                commandFormate: 'main',
                id: 1,
                name: 'mock',
                nodePid: 1,
                paramDesc: 'test',
                purpose: 'forTest',
                resourceId: 1,
                resources: 1,
                taskType: 5,
                udfType: 0,
            });
        });
        expect(onCloseFn).toBeCalled();
    });
});
