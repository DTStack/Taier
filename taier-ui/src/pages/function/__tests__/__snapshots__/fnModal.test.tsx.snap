// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FnModal Component for Function Should match snapshots: create new function 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    新建自定义函数
  </p>
  <form
    autocomplete="off"
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="taskType"
            title="函数类型"
          >
            函数类型
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="taskType_list_0"
                      aria-autocomplete="list"
                      aria-controls="taskType_list"
                      aria-haspopup="listbox"
                      aria-owns="taskType_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="taskType"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="未知"
                  >
                    未知
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="name"
            title="函数名称"
          >
            函数名称
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="name"
                placeholder="请输入函数名称"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="className"
            title="类名"
          >
            类名
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="className"
                placeholder="请输入类名"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            title="资源"
          >
            资源
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                data-testid="mockFolderPicker"
                id="resourceId"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class=""
            for="purpose"
            title="用途"
          >
            用途
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="purpose"
                placeholder="用途"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="commandFormate"
            title="命令格式"
          >
            命令格式
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="commandFormate"
                placeholder="命令格式"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class=""
            for="paramDesc"
            title="参数说明"
          >
            参数说明
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="paramDesc"
                placeholder="请输入函数的参数说明"
                rows="4"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="nodePid"
            title="选择存储位置"
          >
            选择存储位置
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                data-testid="mockFolderPicker"
                id="nodePid"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    data-testid="antd-mock-Modal-confirm"
  >
    confirm
  </button>
</DocumentFragment>
`;

exports[`Test FnModal Component for Function Should match snapshots: edit function 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    编辑自定义函数
  </p>
  <div
    class="task_offline_message"
  >
    <span
      aria-label="exclamation-circle"
      class="anticon anticon-exclamation-circle"
      role="img"
      style="margin-right: 7px;"
    >
      <svg
        aria-hidden="true"
        data-icon="exclamation-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
        />
      </svg>
    </span>
    替换资源时，如果资源的新文件与现有文件名称保持一致，那么替换后关联函数对应任务可立即生效，否则关联函数对应任务需重新提交才可生效。
  </div>
  <form
    autocomplete="off"
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="taskType"
            title="函数类型"
          >
            函数类型
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-status-success ant-select-single ant-select-show-arrow ant-select-disabled"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="taskType_list_0"
                      aria-autocomplete="list"
                      aria-controls="taskType_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="taskType_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      disabled=""
                      id="taskType"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="1"
                  >
                    1
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="name"
            title="函数名称"
          >
            函数名称
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input ant-input-disabled ant-input-status-success"
                disabled=""
                id="name"
                placeholder="请输入函数名称"
                type="text"
                value="test"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="className"
            title="类名"
          >
            类名
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="className"
                placeholder="请输入类名"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            title="资源"
          >
            资源
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                data-testid="mockFolderPicker"
                id="resourceId"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class=""
            for="purpose"
            title="用途"
          >
            用途
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="purpose"
                placeholder="用途"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="commandFormate"
            title="命令格式"
          >
            命令格式
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="commandFormate"
                placeholder="命令格式"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class=""
            for="paramDesc"
            title="参数说明"
          >
            参数说明
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="paramDesc"
                placeholder="请输入函数的参数说明"
                rows="4"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="nodePid"
            title="选择存储位置"
          >
            选择存储位置
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                data-testid="mockFolderPicker"
                id="nodePid"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    data-testid="antd-mock-Modal-confirm"
  >
    confirm
  </button>
</DocumentFragment>
`;
